### ISSUE TITLE
Instability and very high RAM usage when using global statistics in mania

### ISSUE THREAD

**Issue by Joppe27** [ppy/osu#9708]
When using the global statistics while playing mania, there are a lot of stutters and frame drops, extremely high RAM usage and game instability. I can't check for issues in my performance log since it's 1.6 GB because of this bug and notepad doesn't like that. I don't seem to have this issue while playing osu!standard.

https://youtu.be/QjSQNLuKJJs (the RAM does seem to clear at the end of the video causing a massive frame drop, it does go up again after that if I play more mania)

version: 2020.725.0

logs: https://www112.zippyshare.com/v/BnkTkIiU/file.html

**Comment by bdach** [comment]
Global statistics aren't really meant for general usage. They're useful for diagnosing performance issues, which is part of the reason why they are allocating memory (as they are collecting additional diagnostic data).

Playing with them on is not advised unless you're diagnosing an issue yourself or get a request from one of the maintainers to attach a video with them on for an issue.

**Comment by Joppe27** [comment]
Yes! I was trying to find an issue for occasional stutters but that gets kind of hard when the statistics themselves cause much more issues... I didn't see this happen on standard which is why I thought this was a bug. Sorry!

**Comment by bdach** [comment]
As an alternative, you could try using the expanded frame statistics display (press <kbd>Ctrl</kbd>+<kbd>F11</kbd> twice if FPS counter is disabled and once if it is on). I believe they could provide some insight while having a lesser performance impact.

**Comment by smoogipoo** [comment]
This is definitely something that should be supported. Global statistics should not be leaking memory like that.
